packages/components/CONTRIBUTING.md

```diff
@@ -4,12 +4,134 @@ Thank you for taking the time to contribute.
 
 The following is a set of guidelines for contributing to the `@wordpress/components` package to be considered in addition to the general ones described in our [Contributing Policy](/CONTRIBUTING.md).
 
-## Examples
+## Core principles
 
-Each component needs to include an example in its README.md file to demonstrate the usage of the component.
+Contributions to the `@wordpress/components` package should follow a set of core principles and technical requirements.
 
-These examples can be consumed automatically from other projects in order to visualize them in their documentation. To ensure these examples are extractable, compilable and renderable, they should be structured in the following way:
+This set of guidelines should apply especially to newly introduced components. It is, in fact, possible that some of the older component don't respect some of these guidelines for legacy/compat reasons.
 
--   It has to be included in a `jsx` code block.
--   It has to work out-of-the-box. No additional code should be needed to have working the example.
--   It has to define a React component called `My<ComponentName>` which renders the example (i.e.: `MyButton`). Examples for the Higher Order Components should define a `MyComponent<ComponentName>` component (i.e.: `MyComponentWithNotices`).
+### Compatibility
+
+The `@wordpress/components` package includes several components which are relied upon by many developers across different projects. It is, therefore, very important to avoid introducing breaking changes as much as possible.
```

Comment:
I don't think we want to have any breaking changes for stable APIs:

```suggestion
The `@wordpress/components` package includes components that are relied upon by many developers across different projects. It is, therefore, very important to avoid introducing breaking changes.
```

The suggested alternative to soft-deprecate API is the way to go.

Reply:
I know that this PR is not the right place for this conversation to happen, but in the future I'd like to discuss how we can prepare a strategy for introducing breaking changes to this package.

For now, we can definitely go with what you're advocating (thank you for the suggestion!)